hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileCreateHandler.java

```diff
@@ -49,7 +85,76 @@ public void preHandle(@NonNull final PreHandleContext context) {
      *
      * @throws NullPointerException if one of the arguments is {@code null}
      */
-    public void handle() {
-        throw new UnsupportedOperationException("Not implemented");
+    public void handle(
+            @NonNull final HandleContext handleContext,
+            @NonNull final FileCreateTransactionBody fileCreateTransactionBody,
+            @NonNull final FileServiceConfig fileServiceConfig,
+            @NonNull final CreateFileRecordBuilder recordBuilder,
+            @NonNull final WritableFileStoreImpl fileStore) {
+
+        final var builder = new File.Builder();
+
+        if (fileCreateTransactionBody.hasKeys()) {
+            builder.keys(fileCreateTransactionBody.keys());
+        }
+
+        /* Validate if the current file can be created */
+        if (fileStore.sizeOfState() >= fileServiceConfig.maxNum()) {
+            throw new HandleException(MAX_ENTITIES_IN_PRICE_REGIME_HAVE_BEEN_CREATED);
+        }
+
+        final var entityExpiryMeta = new ExpiryMeta(
+                fileCreateTransactionBody.hasExpirationTime()
```

Comment:
Might read clearer if this ternary was assigned to a variable first, so the `new ...` call could be shorter.

Reply:
done